sdk/keyvault/azure-security-keyvault-keys/src/import_key_options.cpp

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include <azure/core/internal/json/json.hpp>
+#include <azure/core/internal/json/json_optional.hpp>
+
+#include <azure/keyvault/common/internal/unix_time_helper.hpp>
+
+#include "azure/keyvault/keys/details/key_constants.hpp"
+#include "azure/keyvault/keys/import_key_options.hpp"
+
+#include <string>
+
+using namespace Azure::Security::KeyVault::Keys;
+using namespace Azure::Security::KeyVault::Keys::_detail;
+using namespace Azure::Core::Json::_internal;
+using namespace Azure::Security::KeyVault::Common::_internal;
+
+std::string
+Azure::Security::KeyVault::Keys::_detail::ImportKeyOptionsSerializer::ImportKeyOptionsSerialize(
+    ImportKeyOptions const& importKeyOptions)
+{
+
+  Azure::Core::Json::_internal::json payload;
+  // key
+  payload[_detail::KeyPropertyName] = importKeyOptions.Key;
+
+  // hsm
+  SetFromNullable(importKeyOptions.HardwareProtected, payload, _detail::HsmPropertyName);
```

Comment:
The implementation changes when it comes to Nullable<T>, and we don't need a function to set a non-nullable<T> value.
For setting anything non-nullable, we can just do
`payload["keyName"] = value`
But for nullable<T>, we need to do
```
if (nullT) {
 payload["KeyName"] = nullT.GetValue();
}
```

I created this method to avoid adding the if all over the place for Nullable<T> objects